### DISCUSSION TITLE
How to add a custom attribute to entities in the knowledge base?

### DISCUSSION THREAD

**Discussion by amirj** [explosion/spaCy#8068] [Help: Best practices]
Is it possible to add a custom attribute when [adding a new entity to the knowledge base](https://spacy.io/api/kb#add_entity)? e.g. a boolean field that could be accessed via entity linker.

**Comment by svlandeg** [comment]
Hi!

I'm afraid that's currently not provided in the built-in implementation, which is tailored towards memory/disk efficiency as knowledge bases tend to grow quite large. I think you have two options to work around this:
- Implement a custom `KnowledgeBase` by extending the provided one.
- Store the additional information in a separate data structure / database that you query independently of the NEL/KB functionality

For future reference, a related feature request is this one: https://github.com/explosion/spaCy/issues/4570
